polyhook2/Detour/ADetour.hpp

```diff
@@ -149,12 +149,10 @@ PLH::insts_t PLH::Detour::relocateTrampoline(insts_t& prologue, uint64_t jmpTblS
 		if (std::find(instsNeedingEntry.begin(), instsNeedingEntry.end(), inst) != instsNeedingEntry.end()) {
 			assert(inst.hasDisplacement());
 			// make an entry pointing to where inst did point to
-			auto entry = makeJmp(jmpTblCurAddr, inst.getDestination());
+			auto entry = makeJmp(jmpTblCurAddr, inst);
 			
-			// move inst to trampoline and point instruction to entry
-			inst.setAddress(inst.getAddress() + delta);
-			inst.setDestination(jmpTblCurAddr);
-			jmpTblCurAddr += jmpSz;
+			if(sizeof(void*) == 4 || !inst.isCalling()) //x64-call instruction does not need a JMP (only needs a dest-holder)
```

Comment:
Polyhook tries to support cross architecture hooking. This pointer size check will break, please check using the mode of the decompiler object

Reply:
fixed in my latest commit